### PR TITLE
docs: update readme

### PR DESCRIPTION
**Pull request by KazuCocoa**
I think we can address iOS 15 specific behavior in https://github.com/appium/appium/issues/15898#issuecomment-927340411 as this readme.
This PR also address go-ios and tidevice as same as https://appium.io/docs/en/advanced-concepts/wda-custom-server/

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,7 +52,9 @@ See [real device configuration documentation](docs/real-device-config.md).
 
 ### Known problems
 
-After many failures on real devices it could transition to a state where connections are no longer being accepted. To possibly remedy this issue reboot the device. Read https://github.com/facebook/WebDriverAgent/issues/507 for more details.
+- After many failures on real devices it could transition to a state where connections are no longer being accepted. To possibly remedy this issue reboot the device. Read https://github.com/facebook/WebDriverAgent/issues/507 for more details.
+- iPhone/iPad real devices show overlay, which has `Automation Running Hold both volume buttons to stop` text, since iOS/iPadOS 15. It is limitation by XCTest framework. It should not affect taking screenshot API in Appium/WebDriverAgent.
+- iPhone/iPad real devices [requires passcode or touch id](https://github.com/appium/appium/issues/15898#issuecomment-927340411) when they start a XCTest session since iOS/iPadOS 15. No passcode/touch id preference can avoid the behavior.
```
Comment (mykola-mokhnach) [inline README.md:57]: requires -> require

File: README.md [hunk 2]
```diff
@@ -53,7 +52,9 @@ See [real device configuration documentation](docs/real-device-config.md).
 
 ### Known problems
 
-After many failures on real devices it could transition to a state where connections are no longer being accepted. To possibly remedy this issue reboot the device. Read https://github.com/facebook/WebDriverAgent/issues/507 for more details.
+- After many failures on real devices it could transition to a state where connections are no longer being accepted. To possibly remedy this issue reboot the device. Read https://github.com/facebook/WebDriverAgent/issues/507 for more details.
+- iPhone/iPad real devices show overlay, which has `Automation Running Hold both volume buttons to stop` text, since iOS/iPadOS 15. It is limitation by XCTest framework. It should not affect taking screenshot API in Appium/WebDriverAgent.
```
Comment (mykola-mokhnach) [inline README.md:56]: This is a known limitation of XCTest framework. This limitation does not affect screenshooting APIs though (e.g. the overlay is not visible on taken screenshots).

File: README.md (hunk 1 above)
Comment (mykola-mokhnach) [reply README.md:57]: "No passcode/touch id" preference allows to avoid the behaviour above

Comment (KazuCocoa) [reply README.md]: 8292934e80109dc65c55dc7da839100be604311d

File: README.md (hunk 2 above)
Comment (KazuCocoa) [reply README.md]: 8292934e80109dc65c55dc7da839100be604311d

**Review by mykola-mokhnach — APPROVED** [review]

**Pull request merged**